src/lib/tls/asio/asio_stream.h

```diff
@@ -546,8 +567,27 @@ class Stream
                return true;
                }
 
+            void tls_verify_cert_chain(
+               const std::vector<X509_Certificate>& cert_chain,
+               const std::vector<std::shared_ptr<const OCSP::Response>>& ocsp_responses,
+               const std::vector<Certificate_Store*>& trusted_roots,
+               Usage_Type usage,
+               const std::string& hostname,
+               const TLS::Policy& policy) override
+               {
+               if(m_tls_context.has_verify_callback())
```

Comment:
Regarding my comment above: We should probably mutex this to ensure that the `m_tls_context` is not altered while performing a handshake.